### PR TITLE
Create Python virtual environment when Open RV/RV is being built

### PR DESCRIPTION
**Pull request by chxmberland**
### Linked issues

N/A

### Summarize your change.

The Python version included with MacOS Sonoma 14.5 adopted [PEP 668](https://peps.python.org/pep-0668/), which is meant to prevent conflicts between system level Python packages with project specific Python packages, like what is included in `requirements.txt`. 

This means that the user will encounter an error when attempting to install Python requirements unless they use the tag `--break-system-packages` tag. By using a virtual environment, system-level conflicts are avoided and the user has a safer build of Open RV. 

### Describe what you have tested and on which operating system.

Mac OS Sonoma 14.5 and Windows 11.

### If possible, provide screenshots.

PEP 668 error without a virtual environment, triggered by running `rvbuild`.

image

### CODE REVIEW

File: rvcmds.sh [hunk 1]
```diff
@@ -79,17 +79,18 @@ RV_BUILD_PARALLELISM="${RV_BUILD_PARALLELISM:-$(python3 -c 'import os; print(os.
...
-alias rvcfgd="cmake -B ${RV_BUILD} -G \"${CMAKE_GENERATOR}\" ${CMAKE_WIN_ARCH} -DCMAKE_BUILD_TYPE=Debug -DRV_DEPS_QT5_LOCATION=${QT_HOME} -DRV_DEPS_WIN_PERL_ROOT=${WIN_PERL}"
-alias rvbuildt="cmake --build ${RV_BUILD} --config Release -v --parallel=${RV_BUILD_PARALLELISM} --target "
-alias rvbuildtd="cmake --build ${RV_BUILD} --config Debug -v --parallel=${RV_BUILD_PARALLELISM} --target "
-alias rvbuild="rvbuildt main_executable"
-alias rvbuildd="rvbuildtd main_executable"
-alias rvtest="ctest --test-dir ${RV_BUILD} --extra=verbose"
-alias rvinst="cmake --install ${RV_BUILD} --prefix ${RV_INST} --config Release"
-alias rvinstd="cmake --install ${RV_BUILD} --prefix ${RV_INST} --config Debug"
-alias rvclean="rm -rf ${RV_BUILD}"
+alias rvenv="[ -d "_virtualenv" ] && source _virtualenv/bin/activate || (python3 -m venv _virtualenv && source _virtualenv/bin/activate)"
```
Comment (eloisebrosseau) [inline rvcmds.sh:82]: The rvenv command does not activate the virtual environment the first time it is run. Have you try to use a function instead like so:
```rvenv() {
if [ -d ".venv" ]; then
    source .venv/bin/activate
else
    python3 -m venv .venv
    source .venv/bin/activate
fi
}
```

Removing the rvenv alias and doing it this way instead fixed the issue for me

**Review by eloisebrosseau — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,4 @@ cmake-*/*
 build/*
 _build/*
 _install/*
+_virtualenv/*
```
[the remark this answers is not in the document]
Comment (cedrik-fuoco-adsk) [reply .gitignore:30]: `venv` is the module name, but you can name the folder however you want. In the link, they have an example with a custom name : `python -m venv tutorial-env`

File: rvcmds.sh (hunk 1 above)
Comment (chxmberland) [reply rvcmds.sh:82]: Ah I see, it's activating in a sub shell, that's why it's not working. Thank you!

File: rvcmds.sh [hunk 3]
```diff
@@ -79,17 +89,18 @@ RV_BUILD_PARALLELISM="${RV_BUILD_PARALLELISM:-$(python3 -c 'import os; print(os.
...
+alias rvcfg="rvenv && cmake -B ${RV_BUILD} -G \"${CMAKE_GENERATOR}\" ${CMAKE_WIN_ARCH} -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_QT5_LOCATION=${QT_HOME} -DRV_DEPS_WIN_PERL_ROOT=${WIN_PERL}"
+alias rvcfgd="rvenv && cmake -B ${RV_BUILD} -G \"${CMAKE_GENERATOR}\" ${CMAKE_WIN_ARCH} -DCMAKE_BUILD_TYPE=Debug -DRV_DEPS_QT5_LOCATION=${QT_HOME} -DRV_DEPS_WIN_PERL_ROOT=${WIN_PERL}"
+alias rvbuildt="rvenv && cmake --build ${RV_BUILD} --config Release -v --parallel=${RV_BUILD_PARALLELISM} --target "
+alias rvbuildtd="rvenv && cmake --build ${RV_BUILD} --config Debug -v --parallel=${RV_BUILD_PARALLELISM} --target "
+alias rvbuild="rvenv && rvbuildt main_executable"
+alias rvbuildd="rvenv && rvbuildtd main_executable"
+alias rvtest="rvenv && ctest --test-dir ${RV_BUILD} --extra=verbose"
+alias rvinst="rvenv && cmake --install ${RV_BUILD} --prefix ${RV_INST} --config Release"
+alias rvinstd="rvenv && cmake --install ${RV_BUILD} --prefix ${RV_INST} --config Debug"
+alias rvclean="rm -rf ${RV_BUILD} && rm -rf _virtualenv"
```
Comment (cedrik-fuoco-adsk) [inline rvcmds.sh:103]: Should change `_virtualenv` to the new name

File: README.md [hunk 4]
```diff
@@ -67,6 +67,12 @@ pre-commit install
 To clean your build directory and restart from a clean slate, use the `rvclean` common build alias, or delete
 the `_build` folder.
 
+### Using a Python Virtual Environment
+
+Open RV automatically sets up a Python virtual environment when the user bootstraps (`rvbootstrap`), builds (`rvbuild`) or configures (`rvcfg`) their project. After each of these commands, you should see that the executing instance of the terminal is using a virtual environment with the name `_virtualenv`.
```
Comment (cedrik-fuoco-adsk) [inline README.md:72]: Should change `_virtualenv` to the new name

File: rvcmds.sh [hunk 5]
```diff
@@ -71,6 +71,16 @@ else
   echo "Using Qt installation already set at $QT_HOME"
 fi
 
+# Must be executed in a function as it changes the shell environment
+rvenv_shell() {
+  if [ -d ".venv" ]; then
+    source .venv/bin/activate
+  else
+    python3 -m venv _venv
+    source .venv/bin/activate
```
Comment (cedrik-fuoco-adsk) [inline rvcmds.sh:80]: The command should be `python3 -m venv .venv` if you want to create a `.venv` folder.

File: .gitignore [hunk 6]
```diff
@@ -27,3 +27,4 @@ cmake-*/*
 build/*
 _build/*
 _install/*
+_venv/*
```
Comment (eloisebrosseau) [inline .gitignore:30]: This should be changed to `.venv`

**Review by cedrik-fuoco-adsk — APPROVED** [review]

File: docs/build_system/config_macos.md [hunk 7]
```diff
@@ -40,6 +42,8 @@ In that case, you can create one with this command `ln -s python3 $(dirname $(wh
 
 ## Install the python requirements
 
+See the **Using a Python Virtual Environment** in `README.md` in the root directory before running any of the commands in this section. Python packages should be installed inside of a virtual environment. 
+
 Some of the RV build scripts requires extra python packages. They can be installed using the requirements.txt at the root of the repository.
 
 ```bash
```
[the remark this answers is not in the document]
Comment (chxmberland) [reply docs/build_system/config_macos.md:50]: Good catch! Fixed

**Review by eloisebrosseau — APPROVED** [review]

**Pull request merged**